open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/impl/VRPConsentRequestHandler.java

```diff
@@ -0,0 +1,222 @@
+
+        try {
+            Object request = consentManageData.getPayload();
+
+            JSONObject validationResponse = VRPConsentRequestValidator.validateVRPPayload(request);
+
+            if (!(Boolean.parseBoolean(validationResponse.getAsString(ConsentExtensionConstants.IS_VALID)))) {
+                log.error(validationResponse.get(ConsentExtensionConstants.ERRORS));
+                throw new ConsentException((ResponseStatus) validationResponse
+                        .get(ConsentExtensionConstants.HTTP_CODE),
+                        String.valueOf(validationResponse.get(ConsentExtensionConstants.ERRORS)));
+            }
+
+            if (StringUtils.isEmpty(consentManageData.getHeaders()
+                    .get(ConsentExtensionConstants.X_IDEMPOTENCY_KEY))) {
+                log.error(ErrorConstants.IDEMPOTENCY_KEY_NOT_FOUND);
+                throw new ConsentException(ResponseStatus.BAD_REQUEST, ErrorConstants.IDEMPOTENCY_KEY_NOT_FOUND);
+            }
+
+            //Handle payment initiation flows
+            handlePaymentPost(consentManageData, request);
+
+        } catch (ConsentManagementException e) {
+            log.error("Error occurred while handling the initiation request", e);
+            throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR,
+                    ErrorConstants.PAYMENT_INITIATION_HANDLE_ERROR);
+        }
+    }
+
+    /**
+     * This method is responsible for handling the GET request for retrieving consent initiation details.
+     * It validates the consent ID, checks if the consent exists,verifies if the consent belongs to the
+     * client making the request.
+     *
+     * @param consentManageData Object
+     */
+    @Override
+    public void handleConsentManageGet(ConsentManageData consentManageData) {
+
+        String consentId = consentManageData.getRequestPath().split("/")[1];
+        if (ConsentManageUtil.isConsentIdValid(consentId)) {
+            try {
+                ConsentResource consent = ConsentServiceUtil.getConsentService().getConsent(consentId,
+                        false);
+                // Check whether the client id is matching
+                if (!consent.getClientID().equals(consentManageData.getClientId())) {
+                    // Throwing same error as null scenario since client will not be able to identify if consent
+                    // exists if consent does not belong to them
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format("ClientIds missmatch. " +
+                                        "consent client id: %s, consent manage data client id: %s",
+                                consent.getClientID(), consentManageData.getClientId()));
+                    }
+                    //throw new RuntimeException()
+                    throw new ConsentException(ResponseStatus.BAD_REQUEST,
+                            new String("test"));
```

Comment:
Shall we add meaningful text here `new String("test")`

Reply:
Fixed in [df474ab]